File: src/eqn/ast/EqnAstNodeTan.java
package eqn.ast;

import eqn.parser.exception.EqnException;
import org.jetbrains.annotations.NotNull;

public class EqnAstNodeTan extends EqnAstNodePreDefinedUnaryFunction {

    public EqnAstNodeTan(@NotNull EqnAstNode equationNode) throws EqnException {
        super("tan", equationNode);
    }

    @Override
    public double evaluate() throws EqnException {
        return Math.tan(operand().evaluate());
    }
}

File: src/eqn/ast/EqnAstNodeTan.kt
package eqn.ast

import eqn.ast.base.EqnAstNode
import eqn.ast.base.EqnAstNodeUnary
import eqn.parser.exception.EqnException
import kotlin.math.tan

class EqnAstNodeTan(operand: EqnAstNode) : EqnAstNodeUnary("tan", Type.PreDefinedUnaryFunction, PrecedenceType.Function, operand) {
    @Throws(EqnException::class)
    override fun evaluate(arguments: Map<String, Double>?): Double {
        return tan(operand.evaluate(arguments))
    }
}